src/bin/main.rs

```diff
@@ -136,12 +172,51 @@ fn check(args: &ArgMatches) -> HowserResult<Option<ValidationProblem>> {
     }
 }
 
+fn process_pharmacy_file(args: &ArgMatches) -> HowserResult<Vec<ValidationProblem>> {
```

Comment:
I recommend reducing the spread of `ArgMatches` as much as reasonably possible, since it represents a squishy bag of strings and encourages redundant presence-checking and interpretation.  The most straightforward remediation is to do immediate local extraction and parsing of relevant parameters as part of the command-flow matching and pass those near-primitives to the "real work" sort of functions.  

A slightly more robust but time-consuming approach would be to parse the contents of the args entirely to some properly-typed structure (either manually or via `structopt`), and dole out subsections as needed.

This is not a blocking critique, just a recommendation.